go.mod

```diff
@@ -1,6 +1,7 @@
 module github.com/ossf/scorecard/v5
 
 go 1.21.12
+toolchain go1.22.5
```

Comment:
This is unfortunately a [dependabot bug](https://github.com/dependabot/dependabot-core/issues/9527). Normally we see this with the osv-scanner updates and I just fix it, but since this requires bumping from 1.21 to 1.22 which has a language behavior change, I opened a separate PR for that (#4322)